magento/utils/transform.ts

```diff
@@ -77,26 +88,65 @@ export const toProduct = (
 
 export const toOffer = (
   { price_info, extension_attributes, sku, currency_code }: MagentoProduct,
+  minInstallmentValue: number,
+  maxInstallments: number
 ): Offer[] => {
   if (!price_info) {
     return [];
   }
-  const { final_price } = price_info;
+
+  const { final_price, max_price, max_regular_price } = price_info;
   const { stock_item } = extension_attributes;
+  const possibleInstallmentsQtd =
+    Math.floor(final_price / minInstallmentValue) || 1;
+  const installments = Array.from(
+    {
+      length: Math.min(possibleInstallmentsQtd, maxInstallments),
+    },
+    (_v, i) => +(final_price / (i + 1)).toFixed(2)
+  );
 
-  const offer: Offer = {
-    "@type": "Offer",
-    availability: stock_item?.is_in_stock ? IN_STOCK : OUT_OF_STOCK,
-    inventoryLevel: {
-      value: stock_item?.qty ?? 0,
+  const priceSpecification: UnitPriceSpecification[] = [
+    {
+      "@type": "UnitPriceSpecification",
+      priceType: "https://schema.org/ListPrice",
+      price: max_price ?? max_regular_price,
     },
-    itemCondition: "https://schema.org/NewCondition",
-    price: final_price,
-    priceCurrency: currency_code,
-    priceSpecification: [],
-    sku: sku,
-  };
-  return [offer];
+    {
+      "@type": "UnitPriceSpecification",
+      priceType: "https://schema.org/SalePrice",
+      price: final_price,
+    },
+    ...installments.map<UnitPriceSpecification>((value, i) => {
+      const [description, billingIncrement] = !i
+        ? ["À vista", final_price]
+        : [i + 1 + "x sem juros", value];
+      return {
+        "@type": "UnitPriceSpecification",
+        priceType: "https://schema.org/SalePrice",
+        priceComponentType: "https://schema.org/Installment",
+        description,
+        billingDuration: i + 1,
+        billingIncrement,
+        price: final_price,
+      };
+    }),
+  ];
+
+  return [
+    {
+      "@type": "Offer",
+      availability: stock_item?.is_in_stock ? IN_STOCK : OUT_OF_STOCK,
+      inventoryLevel: {
+        value: stock_item?.qty ?? 0,
+      },
```

Comment:
precisa de uma validação aqui, tem muito produto com stock_item?.is_in_stock: true so que o  value: stock_item?.qty: 0, precisar validar se ta em Stock e o valor for 0 coloque 999 se o qty tem valor maior que 0 coloque ele